x-pack/plugins/apm/server/lib/services/get_services/get_service_transaction_stats.ts

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  AGENT_NAME,
+  SERVICE_ENVIRONMENT,
+  SERVICE_NAME,
+  TRANSACTION_TYPE,
+} from '../../../../common/elasticsearch_fieldnames';
+import {
+  TRANSACTION_PAGE_LOAD,
+  TRANSACTION_REQUEST,
+} from '../../../../common/transaction_types';
+import { rangeFilter } from '../../../../common/utils/range_filter';
+import { AgentName } from '../../../../typings/es_schemas/ui/fields/agent';
+import {
+  getDocumentTypeFilterForAggregatedTransactions,
+  getProcessorEventForAggregatedTransactions,
+  getTransactionDurationFieldForAggregatedTransactions,
+} from '../../helpers/aggregated_transactions';
+import { getBucketSize } from '../../helpers/get_bucket_size';
+import {
+  calculateTransactionErrorPercentage,
+  getOutcomeAggregation,
+} from '../../helpers/transaction_error_rate';
+import { ServicesItemsSetup } from './get_services_items';
+
+interface AggregationParams {
+  setup: ServicesItemsSetup;
+  searchAggregatedTransactions: boolean;
+}
+
+const MAX_NUMBER_OF_SERVICES = 500;
+
+function calculateAvgDuration({
+  value,
+  deltaAsMinutes,
+}: {
+  value: number;
+  deltaAsMinutes: number;
+}) {
+  return value / deltaAsMinutes;
+}
+
+export async function getServiceTransactionStats({
+  setup,
+  searchAggregatedTransactions,
+}: AggregationParams) {
+  const { apmEventClient, start, end, esFilter } = setup;
+
+  const count = {
+    value_count: {
+      field: getTransactionDurationFieldForAggregatedTransactions(
+        searchAggregatedTransactions
+      ),
+    },
+  };
+
+  const outcomes = getOutcomeAggregation({ searchAggregatedTransactions });
```

Comment:
Are we doing a double `value_count` aggregation now?

Reply:
indeed we are, I'm not sure why the count is needed in the terms outcomes aggregation, do you know?

Reply:
Do you think the count for the outcomes aggregation is relevant to this query?